src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,218 @@
+            {
+                return equalsOrNotEqualsOperation.OperatorKind is BinaryOperatorKind.NotEquals ?
+                    generator.LogicalNotExpression(stringEqualsInvocationExpression) :
+                    stringEqualsInvocationExpression;
+            }
+        }
+
+        /// <summary>
+        /// Replaces <see cref="string.Compare(string, string)"/> violations.
+        /// </summary>
+        private sealed class StringStringCaseReplacer : OperationReplacer
+        {
+            public StringStringCaseReplacer(RequiredSymbols symbols)
+                : base(symbols)
+            { }
+
+            public override bool IsMatch(IBinaryOperation violation) => UseStringEqualsOverStringCompare.IsStringStringCase(violation, Symbols);
+
+            public override SyntaxNode CreateReplacementExpression(IBinaryOperation violation, SyntaxGenerator generator)
+            {
+                RoslynDebug.Assert(IsMatch(violation));
+
+                var compareInvocation = GetInvocation(violation);
+                var equalsInvocationSyntax = generator.InvocationExpression(
+                    CreateEqualsMemberAccess(generator),
+                    compareInvocation.Arguments.GetArgumentsInParameterOrder().Select(x => x.Value.Syntax));
+
+                return InvertIfNotEquals(equalsInvocationSyntax, violation, generator);
+            }
+        }
+
+        /// <summary>
+        /// Replaces <see cref="string.Compare(string, string, bool)"/> violations.
+        /// </summary>
+        private sealed class StringStringBoolReplacer : OperationReplacer
+        {
+            public StringStringBoolReplacer(RequiredSymbols symbols)
+                : base(symbols)
+            { }
+
+            public override bool IsMatch(IBinaryOperation violation) => UseStringEqualsOverStringCompare.IsStringStringBoolCase(violation, Symbols);
+
+            public override SyntaxNode CreateReplacementExpression(IBinaryOperation violation, SyntaxGenerator generator)
+            {
+                RoslynDebug.Assert(IsMatch(violation));
+
+                var compareInvocation = GetInvocation(violation);
+
+                //  We know that the 'ignoreCase' argument in 'string.Compare(string, string, bool)' is a boolean literal
+                //  because we've asserted that 'IsMatch' returns true.
+                var ignoreCaseLiteral = (ILiteralOperation)compareInvocation.Arguments.GetArgumentForParameterAtIndex(2).Value;
+
+                //  If the violation contains a call to 'string.Compare(x, y, true)' then we
+                //  replace it with a call to 'string.Equals(x, y, StringComparison.CurrentCultureIgnoreCase)'.
+                //  If the violation contains a call to 'string.Compare(x, y, false)' then we
+                //  replace it with a call to 'string.Equals(x, y, StringComparison.CurrentCulture)'. 
+                var stringComparisonEnumMemberName = (bool)ignoreCaseLiteral.ConstantValue.Value ?
```

Comment:
This should likely be `Ordinal` and `OrdinalIgnoreCase` based on [guidelines](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings)

Reply:
@ryzngard string.Compare is using CurrentCulture by default.

https://source.dot.net/#System.Private.CoreLib/String.Comparison.cs,200

So probably the codefix would want to keep the *same* semantics.

Reply:
I don't fully agree. In general we don't want to change explicit semantics for code, but there's no way to know if the user intended to use `CurrentCulture` or not. When it's ambiguous, I think suggesting the public documented guidelines would be a good thing.

Reply:
The examples in [the issue](https://github.com/dotnet/runtime/issues/45552) specified that we use `CurrentCulture` and `CurrentCultureIgnoreCase` so we don't change semantics.

Reply:
I don't see any explicit discussion in the issue about whether the replacement for `string.Compare(s1, s2)` should use `CurrentCulture` or `Ordinal`. That said, I'm not strong enough of the opinion that we should change the functional meaning for potential correction. Especially if others think it should be exact parity.  

I will expand on my thought process for discussion, but consider this non-blocking: 

https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings specifies
> "Use overloads that explicitly specify the string comparison rules for string operations. Typically, this involves calling a method overload that has a parameter of type StringComparison." 

If I were a user trying to update code to follow that guideline, how would I consider doing it? I think most cases I would add a `Ordinal` or `OrdinalIgnoreCase` as the intent for string comparison unless context dictates I should be using culture aware comparison. That also follows the guidance of 
> "Use StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase for comparisons as your safe default for culture-agnostic string matching." 

and 

> "Use comparisons with StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase for better performance."

This fix is listed as `Usage`, but the key findings in the issue are centered around performance. Probably because nobody disagreed that `string.Equals(...)` is more readable than `string.Compare(...) == 0`. 

If we are encouraging users to have mostly correct and performant code, or even for what I would consider to give as guidance as "probably correct" in most cases, we would use `Ordinal` or `OrdinalIgnoreCase`.

Reply:
In my opinion, a code fix should only change the issue that its analyzer is reporting on. This analyzer is not flagging incorrect uses of `CurrentCulture`, it is flagging uses of a total-order comparison when a simple equality check was desired. I would not be opposed to an analyzer that flags usages of any `Compare` or `Equals` overload that doesn't specify the kind of comparison, but I don't think this analyzer should be changing semantics.